radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticlesMapper.java

```diff
@@ -15,52 +17,64 @@
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 import java.util.function.Supplier;
+import java.util.stream.Stream;
 
 /**
  * Maps a send message action to the particles necessary to be included in an atom.
  */
 public class SendMessageToParticlesMapper implements ActionToParticlesMapper {
+
+	/**
+	 * A module capable of creating new securely random ECKeyPairs
+	 */
 	private final Supplier<ECKeyPair> generator;
 
+	/**
+	 * Function describing given an action who do we encrypt a message for
+	 */
+	private final Function<SendMessageAction, Stream<ECPublicKey>> encryptionScheme;
+
 	public SendMessageToParticlesMapper(Supplier<ECKeyPair> generator) {
+		this(generator, sendMsg -> Stream.of(sendMsg.getFrom(), sendMsg.getTo()).map(RadixAddress::getPublicKey));
+	}
+
+	public SendMessageToParticlesMapper(Supplier<ECKeyPair> generator, Function<SendMessageAction, Stream<ECPublicKey>> encryptionScheme) {
```

Comment:
Funny how the private member fields were JavaDoc-ed, but these public methods were not :)

Reply:
It's hilarious!